### PR TITLE
CREATE TABLE multi-region syntax updates (LOCALITY)

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes https://github.com/cockroachdb/docs/issues/10088.
Fixes https://github.com/cockroachdb/docs/issues/9554.
Fixes https://github.com/cockroachdb/docs/issues/9613.
Fixes https://github.com/cockroachdb/docs/issues/9303.

### CODE REVIEW

**Review by ajstorm — CHANGES_REQUESTED** [review]
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10550)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani and @ericharmeling)

___
*[_includes/v21.1/sql/generated/diagrams/opt_locality.html, line 38 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__ff_I9Z4M3Rthd7Rs:-M__ff_I9Z4M3Rthd7Rt:b-s23yd1) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/_includes/v21.1/sql/generated/diagrams/opt_locality.html#L38)):*
> ```HTML
> <rect x="407" y="297" width="56" height="32"></rect>
> <rect x="405" y="295" width="56" height="32" class="nonterminal"></rect>
> <text class="nonterminal" x="415" y="315">name</text></a><rect x="177" y="341" width="36" height="32" rx="10"></rect>
> ```

Nit: Can we make this `column_name`?
___
*[v21.1/create-table.md, line 561 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__gBi6ELWbSldlcbvG:-M__gBi76vFDOiHXYWqb:b-ngwthw) ([raw file](https://github.com/cockroachdb/docs/blob/f896eb7c02279c4ed1dae579714faf312da58f8c/v21.1/create-table.md#L561)):*
> ```Markdown
> (1 row)
> ~~~
> ```

Nice to see this stuff go.
___
*[v21.1/create-table.md, line 678 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__gXk5CNGHzebq9MTG:-M__gXk5CNGHzebq9MTH:b26fan7) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L678)):*
> ```Markdown
> In order to set table localities, the database that contains the table must have [database regions](multiregion-overview.html#database-regions).
> 
> By default, all tables in a multi-region database have a [`REGIONAL`](multiregion-overview.html#regional-tables) locality.
> ```

Do we want to say REGIONAL BY TABLE IN PRIMARY REGION locality?
___
*[v21.1/create-table.md, line 685 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__gi40Crdm2qEHbgPq:-M__gi40Crdm2qEHbgPr:bbosc7y) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L685)):*
> ```Markdown
> To create a table with a [`GLOBAL`](multiregion-overview.html#global-tables) locality, add a `LOCALITY GLOBAL` clause to the end of the `CREATE TABLE` statement.
> 
> The `GLOBAL` locality is useful for "read-mostly" tables of reference data that are rarely updated, but need to be available to all regions.
> ```

I'd change this to "...but need to be read with low latency from all regions."

All tables are technically available in all regions, it's the latency that GLOBAL tables help optimize.
___
*[v21.1/create-table.md, line 715 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__h1_43MzQ79Tt4Z-D:-M__h1_43MzQ79Tt4Z-E:bos2nu5) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L715)):*
> ```Markdown
> #### Create a table with a regional-by-table locality
> 
> To create a table with a [`REGIONAL BY TABLE`](multiregion-overview.html#regional-tables) locality, add a `LOCALITY REGIONAL BY TABLE` clause to the end of the `CREATE TABLE` statement.
> ```

Nit: We might want to also say that REGIONAL alone denotes REGIONAL BY TABLE.
___
*[v21.1/create-table.md, line 751 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__hw_X8aY9gh8u_UPn:-M__hw_X8aY9gh8u_UPo:bpzgc8x) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L751)):*
> ```Markdown
> To create a table with a [`REGIONAL-BY-ROW`](multiregion-overview.html#regional-by-row-tables) locality, add a `LOCALITY REGIONAL BY ROW` clause to the end of the `CREATE TABLE` statement.
> 
> The `REGIONAL BY ROW` locality is useful for tables that require low-latency reads and writes from different regions, depending on the row in the table.
> ```

Nit: Can we massage this to "...from different regions, where the low-latency region is specified at the row level"
___
*[v21.1/create-table.md, line 770 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__iYFtET1nONGz-zRi:-M__iYFtET1nONGz-zRj:b-zfnj7s) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L770)):*
> ```Markdown
> ~~~
> 
> CockroachDB will automatically assign rows to regions, based on the locality of the node from which a row is inserted. It will then optimize all read queries on the table, based on the locality of the node.
> ```

The last sentence here could be a bit misleading.  I'd reword to "It will then optimize subsequent reads/writes of that row from the inserting region."
___
*[v21.1/create-table.md, line 772 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__kHqCAdjmFZhD_T04:-M__kHqCAdjmFZhD_T05:b-sx7qtp) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L772)):*
> ```Markdown
> CockroachDB will automatically assign rows to regions, based on the locality of the node from which a row is inserted. It will then optimize all read queries on the table, based on the locality of the node.
> 
> To override CockroachDB's automatic region assignment, you can assign rows to regions based on the value of a column of type `crdb_internal_region`.
> ```

I'd also reword this slightly:

"To assign rows to regions CockroachDB maintains a hidden crdb_region column.  To override the automatic region assignment and choose the region in which rows will be placed, you can provide a value for the crdb_region column at insert (or update) time.  The region value assigned must be one of the regions added to the database, and present in the crdb_internal_region enum."
___
*[v21.1/create-table.md, line 775 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__l1RA0bUOYg-U2AWQ:-M__l1RA0bUOYg-U2AWR:b-quyiao) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L775)):*
> ```Markdown
> 
> For example:
> ```

I'd also change this example so that it shows the default case (not REGIONAL BY ROW AS).

**Review by arulajmani — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10550)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm and @ericharmeling)

___
*[v21.1/create-table.md, line 678 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__gXk5CNGHzebq9MTG:-M_a4Q8z81UhFuED_wUp:b23tukp) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L678)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

Do we want to say REGIONAL BY TABLE IN PRIMARY REGION locality?
</blockquote></details>

+1, we should mention that the table will be localized in the primary region in the default case. 
___
*[v21.1/create-table.md, line 715 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__h1_43MzQ79Tt4Z-D:-M_a4nvQ3p8DwmeMVCzE:br7sfjt) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L715)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

Nit: We might want to also say that REGIONAL alone denotes REGIONAL BY TABLE.
</blockquote></details>

Not sure which syntax we're promoting, but would it make sense to change these examples to say `REGIONAL IN ...` instead of `REGIONAL BY TABLE IN ...` as well?
___
*[v21.1/create-table.md, line 770 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_a7S-g0okzppL-xzP1:-M_a7S-g0okzppL-xzP2:byzk3om) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L770)):*
> ```Markdown
> ~~~
> 
> CockroachDB will automatically assign rows to regions, based on the locality of the node from which a row is inserted. It will then optimize all read queries on the table, based on the locality of the node.
> ```

Not sure if it is information overload, but if you think it's worth mentioning, you could add that if the node from which a row is inserted is in a region that does not belong to the database, then the row will be homed in the primary region. 
___
*[v21.1/create-table.md, line 775 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__l1RA0bUOYg-U2AWQ:-M_a5Y2v30FaMFXLkt_P:b-fmmp9d) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L775)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

I'd also change this example so that it shows the default case (not REGIONAL BY ROW AS).
</blockquote></details>

Do we not want to promote REGIONAL BY ROW AS? I agree that we should document your suggestion above, that `crdb_region` can be inserted into and updated,  and that will affect the rows homing. However, I do think there is some value in documenting the REGIONAL BY ROW AS syntax with a motivating example. 

Maybe a computed column? In case people want the flexibility. 
___
*[v21.1/create-table.md, line 792 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_a6mRa9p0JrSrBXAdX:-M_a6mRb-xwQseAxNKLS:b-ntx3y4) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L792)):*
> ```Markdown
> ~~~
> 
> You can then manually set the values of the region with each [`INSERT`](insert.html) statement:
> ```

I'd move this to the `REGIONAL BY ROW` section, as this is something you can do with the auto-generated `crdb_region` column as well (as Adam noted above)
___
*[v21.1/create-table.md, line 802 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_a77LO2e7e1a-8tnQA:-M_a77LO2e7e1a-8tnQB:by2dpso) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L802)):*
> ```Markdown
> 
> ~~~ sql
> > UPDATE vehicles SET region = "us-east1" WHERE city IN (...) ...
> ```

If we move this in the CREATE TABLE statement itself, it can showcase a compelling use of `REGIONAL BY ROW AS`

**Review by ericharmeling — COMMENTED** [review]
TFTR @ajstorm , @arulajmani 

I think I got to all your comments. PTAL.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10550#-:-M_vPAmN3GDiSROFY0Y3:b-rxzbk9)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm and @arulajmani)

___
*[_includes/v21.1/sql/generated/diagrams/opt_locality.html, line 38 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__ff_I9Z4M3Rthd7Rs:-M_vP9vV62kWQG40SY1C:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/_includes/v21.1/sql/generated/diagrams/opt_locality.html#L38)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

Nit: Can we make this `column_name`?
</blockquote></details>

Done.
___
*[v21.1/create-table.md, line 678 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__gXk5CNGHzebq9MTG:-M_um2WXC1yKcFWUO1SI:b-duzahh) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L678)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

+1, we should mention that the table will be localized in the primary region in the default case. 
</blockquote></details>

Yes! Done.
___
*[v21.1/create-table.md, line 685 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__gi40Crdm2qEHbgPq:-M_umS0k28DHerR_htzu:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L685)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

I'd change this to "...but need to be read with low latency from all regions."

All tables are technically available in all regions, it's the latency that GLOBAL tables help optimize.
</blockquote></details>

Done.
___
*[v21.1/create-table.md, line 715 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__h1_43MzQ79Tt4Z-D:-M_uum_n1WEB1MuIqSgq:bqqcm8n) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L715)):*
> We might want to also say that REGIONAL alone denotes REGIONAL BY TABLE.

Added a note.

>`REGIONAL IN ...` instead of `REGIONAL BY TABLE IN ...`

IMO, `REGIONAL BY TABLE` is easier to distinguish from `REGIONAL BY ROW`.
___
*[v21.1/create-table.md, line 751 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__hw_X8aY9gh8u_UPn:-M_uv6K92HXpK8cJx4WK:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L751)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

Nit: Can we massage this to "...from different regions, where the low-latency region is specified at the row level"
</blockquote></details>

Done.
___
*[v21.1/create-table.md, line 770 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__iYFtET1nONGz-zRi:-M_uvbAy8W7JxF4_LM1P:b28bk3i) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L770)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

The last sentence here could be a bit misleading.  I'd reword to "It will then optimize subsequent reads/writes of that row from the inserting region."
</blockquote></details>

Hmm... You are right - the wording is vague and a little tricky here.

How does this work:

"CockroachDB will automatically assign each row to a region, based on the locality of the node from which the row is inserted. It will then optimize subsequent read and write queries executed from nodes located in the region assigned to the rows being queried."

?
___
*[v21.1/create-table.md, line 770 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_a7S-g0okzppL-xzP1:-M_uvbOmAgAkXmTRJ0mu:b-y5d7ow) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L770)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Not sure if it is information overload, but if you think it's worth mentioning, you could add that if the node from which a row is inserted is in a region that does not belong to the database, then the row will be homed in the primary region. 
</blockquote></details>

That's good information - added.
___
*[v21.1/create-table.md, line 772 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__kHqCAdjmFZhD_T04:-M_v5bIF-ZOHk_Ic45Nr:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L772)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

I'd also reword this slightly:

"To assign rows to regions CockroachDB maintains a hidden crdb_region column.  To override the automatic region assignment and choose the region in which rows will be placed, you can provide a value for the crdb_region column at insert (or update) time.  The region value assigned must be one of the regions added to the database, and present in the crdb_internal_region enum."
</blockquote></details>

Done.
___
*[v21.1/create-table.md, line 775 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M__l1RA0bUOYg-U2AWQ:-M_v5gmV8jh4jd5S0HH7:b-z91jvh) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L775)):*
> I'd also change this example so that it shows the default case (not REGIONAL BY ROW AS).

Done.

> Maybe a computed column? In case people want the flexibility.

I've added a new example subsection that uses `REGIONAL BY ROW AS`. This new example defines a computed `crdb_internal_region`-typed column, based on the `city` value.
___
*[v21.1/create-table.md, line 792 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_a6mRa9p0JrSrBXAdX:-M_vDCnI5rwVDrqvh-Qg:b-34xta) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L792)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

I'd move this to the `REGIONAL BY ROW` section, as this is something you can do with the auto-generated `crdb_region` column as well (as Adam noted above)
</blockquote></details>

See my comment above re: the new example.
___
*[v21.1/create-table.md, line 802 at r1](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_a77LO2e7e1a-8tnQA:-M_vDCL8BLfh7zhD60d4:bm2rs5d) ([raw file](https://github.com/cockroachdb/docs/blob/bcbdcdb2beabc22a3f8a61cfec690ba15fd9ded8/v21.1/create-table.md#L802)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

If we move this in the CREATE TABLE statement itself, it can showcase a compelling use of `REGIONAL BY ROW AS`
</blockquote></details>

Done. (See my comment above re: the new example.)

File: v21.1/create-table.md [hunk 1]
```diff
@@ -764,6 +668,205 @@ Note that the foreign key constraint `fk_owner_id_ref_users` in the source table
...
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE TABLE promo_codes (
+    code STRING PRIMARY KEY,
+    description STRING,
+    creation_time TIMESTAMP,
+    expiration_time TIMESTAMP,
+    rules JSONB)
+    LOCALITY GLOBAL
+  ;
```
Comment (rmloveland) [inline v21.1/create-table.md:698]: nit: move dangling semicolon up?

File: v21.1/create-table.md [hunk 2]
```diff
@@ -764,6 +668,205 @@ Note that the foreign key constraint `fk_owner_id_ref_users` in the source table
...
+For example, suppose you want to create a table for your application's end users in a specific state:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE TABLE users_ny (
+    id UUID PRIMARY KEY,
+    name STRING,
+    address STRING)
+    LOCALITY REGIONAL BY TABLE IN "us-east1"
+  ;
```
Comment (rmloveland) [inline v21.1/create-table.md:732]: move dangling semicolon up?

File: v21.1/create-table.md [hunk 3]
```diff
@@ -764,6 +668,205 @@ Note that the foreign key constraint `fk_owner_id_ref_users` in the source table
...
+    id UUID PRIMARY KEY,
+    type STRING,
+    city STRING,
+    owner_id UUID,
+    creation_time TIMESTAMP,
+    status STRING,
+    current_location STRING,
+    ext JSONB)
+    LOCALITY REGIONAL BY ROW
+  ;
```
Comment (rmloveland) [inline v21.1/create-table.md:771]: nit: move dangling semicolon up?

File: v21.1/create-table.md [hunk 4]
```diff
@@ -764,6 +668,205 @@ Note that the foreign key constraint `fk_owner_id_ref_users` in the source table
...
+~~~
+
+CockroachDB will automatically assign each row to a region, based on the locality of the node from which the row is inserted. It will then optimize subsequent read and write queries executed from nodes located in the region assigned to the rows being queried.
```
Comment (rmloveland) [inline v21.1/create-table.md:774]: nit: don't need comment between "region, based"

File: v21.1/create-table.md [hunk 5]
```diff
@@ -764,6 +668,205 @@ Note that the foreign key constraint `fk_owner_id_ref_users` in the source table
...
+
+{{site.data.alerts.callout_info}}
+If the node from which a row is inserted has a locality that does not correspond to a region in the database, then the row will be assigned to the database's primary region.
+{{site.data.alerts.end}}
+
+To assign rows to regions, CockroachDB creates and manages a hidden `crdb_region` column, of [`ENUM`](enum.html) type `crdb_internal_region`. To override the automatic region assignment and choose the region in which rows will be placed, you can provide a value for the `crdb_region` column in `INSERT` and `UPDATE` queries on the table.
+
+{{site.data.alerts.callout_info}}
+The region value for `crdb_region` must be one of the regions added to the database, and present in the `crdb_internal_region` `ENUM`. To return the available regions, use a [`SHOW REGIONS FROM DATABASE <database name>`](show-regions.html) statement, or a [`SHOW ENUMS`](show-enums.html) statement.
```
Comment (rmloveland) [inline v21.1/create-table.md:783]: Could add some cross-linkage to more docs on `crdb_region` in https://www.cockroachlabs.com/docs/v21.1/set-locality#crdb_region

File: v21.1/create-table.md [hunk 6]
```diff
@@ -764,6 +668,205 @@ Note that the foreign key constraint `fk_owner_id_ref_users` in the source table
...
+    id UUID PRIMARY KEY,
+    type STRING,
+    city STRING,
+    owner_id UUID,
+    creation_time TIMESTAMP,
+    status STRING,
+    current_location STRING,
+    ext JSONB)
+    LOCALITY REGIONAL BY ROW;
+  ;
```
Comment (rmloveland) [inline v21.1/create-table.md:800]: nit: move dangling semicolon up?

File: v21.1/set-locality.md [hunk 7]
```diff
@@ -122,17 +122,12 @@ In fact, you can specify any column definition you like for the `REGIONAL BY ROW
...
-  CASE WHEN city = 'amsterdam' THEN 'europe-west1'
-       WHEN city = 'paris' THEN 'europe-west1'
-       WHEN city = 'rome' THEN 'europe-west1'
-       WHEN city = 'new york' THEN 'us-east1'
-       WHEN city = 'boston' THEN 'us-east1'
-       WHEN city = 'washington dc' THEN 'us-east1'
-       WHEN city = 'san francisco' THEN 'us-west1'
-       WHEN city = 'seattle' THEN 'us-east1'
-       WHEN city = 'los angeles' THEN 'us-east1'
+  CASE
```
Comment (rmloveland) [inline v21.1/set-locality.md:125]: FINE THEN ERIC
FINE THEN
:-)

File: v21.1/set-locality.md [hunk 8]
```diff
@@ -122,17 +122,12 @@ In fact, you can specify any column definition you like for the `REGIONAL BY ROW
...
+    WHEN city IN ('new york', 'boston', 'washington dc', 'chicago', 'detroit', 'minneapolis') THEN 'us-east-1'
+    WHEN city IN ('san francisco', 'seattle', 'los angeles') THEN 'us-west-1'
+    WHEN city IN ('amsterdam', 'paris', 'rome') THEN 'eu-west-1'  
   END
-) STORED
+) STORED;
```
Comment (rmloveland) [inline v21.1/set-locality.md:130]: noice ;-)

**Review by rmloveland — APPROVED** [review]
LGTM, mentioned some small non-blocking things

**Review by ericharmeling — COMMENTED** [review]
TFTR, @rmloveland !

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10550#-:-M_w-ZPS0N4891E0X9-e:b-jr95f4)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm, @arulajmani, and @rmloveland)

___
*[v21.1/create-table.md, line 698 at r2](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_W_rTLcNtp3XImkDfU-r2-698:-M_vzrtTDz5YMzNbI1Ee:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/263a2e1cb637b5bfc12e233f8a2b167973b71fbe/v21.1/create-table.md#L698)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

nit: move dangling semicolon up?
</blockquote></details>

Done.
___
*[v21.1/create-table.md, line 732 at r2](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_W_rTLcNtp3XImkDfU-r2-732:-M_vzvpOC56iW4AVLQ2b:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/263a2e1cb637b5bfc12e233f8a2b167973b71fbe/v21.1/create-table.md#L732)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

move dangling semicolon up?
</blockquote></details>

Done.
___
*[v21.1/create-table.md, line 771 at r2](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_W_rTLcNtp3XImkDfU-r2-771:-M_vzxGUCMU5T1LiQQ3K:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/263a2e1cb637b5bfc12e233f8a2b167973b71fbe/v21.1/create-table.md#L771)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

nit: move dangling semicolon up?
</blockquote></details>

Done.
___
*[v21.1/create-table.md, line 774 at r2](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_W_rTLcNtp3XImkDfU-r2-774:-M_w--E45u4jxuwzPZ4R:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/263a2e1cb637b5bfc12e233f8a2b167973b71fbe/v21.1/create-table.md#L774)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

nit: don't need comment between "region, based"
</blockquote></details>

Done.
___
*[v21.1/create-table.md, line 783 at r2](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_W_rTLcNtp3XImkDfU-r2-783:-M_w-1mU0t0dx_DSus0T:b-7zymug) ([raw file](https://github.com/cockroachdb/docs/blob/263a2e1cb637b5bfc12e233f8a2b167973b71fbe/v21.1/create-table.md#L783)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

Could add some cross-linkage to more docs on `crdb_region` in https://www.cockroachlabs.com/docs/v21.1/set-locality#crdb_region 
</blockquote></details>

Oooo! Did not see this anchor link. Done.
___
*[v21.1/create-table.md, line 800 at r2](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_W_rTLcNtp3XImkDfU-r2-800:-M_w-Pmc0pnWOuKhQyMK:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/263a2e1cb637b5bfc12e233f8a2b167973b71fbe/v21.1/create-table.md#L800)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

nit: move dangling semicolon up?
</blockquote></details>

Done.
___
*[v21.1/set-locality.md, line 125 at r2](https://reviewable.io/reviews/cockroachdb/docs/10550#-M_W_rTLcNtp3XImkDfV-r2-125:-M_w-RQJ3rnEplVZOKS2:bez851s) ([raw file](https://github.com/cockroachdb/docs/blob/263a2e1cb637b5bfc12e233f8a2b167973b71fbe/v21.1/set-locality.md#L125)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

FINE THEN ERIC
FINE THEN
:-) 
</blockquote></details>

:smile_cat:

**Comment by ericharmeling** [comment]
I'm going to go ahead and merge.

@ajstorm @arulajmani Feel free to add more feedback if you like. Happy to make revisions after this is published.

**Review by arulajmani — COMMENTED** [review]
The new changes with `REGIONAL BY ROW AS`... are great! :D
